plugins/inventory/gql_inventory.py

```diff
@@ -186,43 +186,60 @@ def add_variable(self, host: str, var: str, var_type: str):
+
+    def add_ansible_platform(self, device):
+        """Add network platform to host"""
+        if device["platform"] and "napalm_driver" in device["platform"]:
+            self.add_variable(
+                device["name"],
+                ANSIBLE_LIB_MAPPER_REVERSE.get(NAPALM_LIB_MAPPER.get(device["platform"]["napalm_driver"])),  # Convert napalm_driver to ansible_network_os value
+                "ansible_network_os",
+            )
+
+    def create_groups(self, device):
+        """Create groups specified and add device to group."""
+        device_name = device["name"]
+        for group_by_path in self.group_by:
+            parent_attr, *chain = group_by_path.split(".")
+            device_attr = device.get(parent_attr)
+            if device_attr is None:
+                self.display.display(f"Could not find value for {parent_attr} on device {device_name}")
+                continue
+
+            if not chain:
+                group_name = device_attr
+
+            while chain:
+                group_name = chain.pop(0)
+                if isinstance(device_attr.get(group_name), Mapping):
+                    device_attr = device_attr.get(group_name)
                     continue
-                for attribute in group_by_path[1:]:
-                    if not isinstance(attr_value, Mapping):
-                        error_path = group_by.split(attribute)[0][:-1]
-                        self.display.display(f"Device {device_name} attribute {error_path} is not a dictionary.")
-                        continue
+                else:
                     try:
-                        attr_value = attr_value[attribute]
+                        group_name = device_attr[group_name]
                     except KeyError:
-                        self.display.display(f"Could not find value for {attribute} in {group_by} on device {device_name}")
+                        self.display.display(f"Could not find value for {group_name} in {group_by_path} on device {device_name}")
                         break
-                if isinstance(attr_value, Mapping):
-                    try:
-                        attr_value = attr_value["name"]
-                        self.display.display(f"No name value for {attr_value} on device {device_name} for group {group_by}.")
-                    except KeyError:
-                        try:
-                            attr_value = attr_value["slug"]
-                        except KeyError:
-                            self.display.display(f"No slug value for {attr_value} on device {device_name} for group {group_by}.")
-                            continue
-                if isinstance(attr_value, str):
-                    self.inventory.add_group(attr_value)
-                    self.inventory.add_child(attr_value, device_name)
+
+            if isinstance(group_name, Mapping):
+                if "name" in group_name:
```

Comment:
Yes, Jacob and I talked about that too. We think we should update the default graphql query to get slug rather than name. https://github.com/nautobot/nautobot-ansible/blob/develop/plugins/templates/graphql_default_query.j2 Thoughts?

Reply:
I would agree with that change. @jvanderaa What about yourself?

Reply:
I think this may be the one of the last main concerns here. Just pinging @jvanderaa again.

Reply:
If we make that change we have to move to a new version. I think we need to keep it where we are at for now and open an issue. That would be a breaking change.